src/runtime/HalideRuntimeHexagonDma.h

```diff
@@ -0,0 +1,145 @@
+#ifndef HALIDE_HALIDERUNTIMEHEXAGONDMA_H
+#define HALIDE_HALIDERUNTIMEHEXAGONDMA_H
+
+
+#include "HalideRuntime.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum halide_hexagon_dma_user_fmt{
+	NV12,
+	UBWC_NV12,
+	P010,
+	TP10,
+	NV124R,
+	UBWC_NV124R,
+}halide_hexagon_dma_user_fmt_t;
+
+typedef enum halide_hexagon_user_component{
+	LUMA_COMPONENT,                /*frame has only Luma component*/
+	CHROMA_COMPONENT,              /*frame has only Chroma Component*/
+	BOTH_LUMA_CHROMA               /*frame has both Luma nad chroma components */
+
+}halide_hexagon_dma_user_component_t;
+
+/** \file
+ *  Routines specific to the Halide Hexagon dma runtime.
+ */
+
+typedef int halide_hexagon_dma_handle_t;
+
+extern const struct halide_device_interface_t *halide_hexagon_dma_device_interface();
+
+/* CreateContext check if DMA Engines are available
+ * Allocates Memory for a DMA Context
+ * Returns Error if DMA Context is not available
+ */
+extern int halide_hexagon_dmaapp_create_context(void** user_context, int nFrames);
+
+/* Attach Context checks if the frame width, height and type is aligned with DMA Transfer
+ * Returns Error if the frame is not aligned.
+ * AttachContext needs to be called for each frame
+ */
+extern int halide_hexagon_dmaapp_attach_context(void* user_context, addr_t  frame, int type, int d, int w, int h, int s, int last);

```

Comment:
Please fix the indentation here. It isn't good to have long lines of code that wrap around. Consider changing this to
```
extern int halide_hexagon_dmaapp_attach_context(void* user_context, addr_t  frame, int type, 
                                                int d, int w, int h, int s, int last);
```